snuba/web/rpc/v1/endpoint_trace_item_table.py

```diff
@@ -104,4 +109,10 @@ def _execute(self, in_msg: TraceItemTableRequest) -> TraceItemTableResponse:
                 "This endpoint requires meta.trace_item_type to be set (are you requesting spans? logs?)"
             )
         resolver = self.get_resolver(in_msg.meta.trace_item_type)
-        return resolver.resolve(in_msg)
+        try:
+            return resolver.resolve(in_msg)
+        except Exception as e:
+            if "DB::Exception: Memory limit (for query) exceeded" in str(e):
```

Comment:
Should code `241` be used instead of the string matching? That way if ClickHouse changes the exception message, it will still work.